### PR TITLE
Fabo/loading optimization

### LINKED ISSUE #4366: Initial load of vendor dependencies takes too long
**Is your feature request related to a problem? Please describe.**

When Lunie loads, the initial dependencies are too big and therefore it takes long to load them.

**Describe the solution you'd like**

Look to reduce the size of needed dependencies. Investigate tree shaking in webpack.
Load some dependencies asynchronous so Lunie starts quicker while some needed processes get executed later. Example: Load firebase dependencies async and execute the code for authentication asynchronous.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by faboweb**
Closes #4366

Optimize Lunie initial loading by loading some deps async

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5ef5009f87df848a16e96ea8--lunieio.netlify.app

**Comment by faboweb** [comment]
Load time reduction by 5s

**Comment by faboweb** [comment]
All files are reevaluated. So on a refresh all deps are loaded again. Will fix this in another PR.

**Comment by faboweb** [comment]
Looking damn fast

**Comment by faboweb** [comment]
Load time on refresh 2.5s
Finish JS in 5.7s

Before:
Load 21.3s
Finish JS in 25.4s

**Comment by Bitcoinera** [comment]
> Load time on refresh 2.5s
> Finish JS in 5.7s
> 
> On new load
> Load 4.2s
> Finish JS in 8.1s
> 
> Before:
> Load 21.3s
> Finish JS in 25.4s

File: app/package.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "description": "Lunie is the staking and governance platform for proof-of-stake blockchains.",
   "author": "Lunie International Software Systems Inc. <hello@lunie.io>",
   "scripts": {
-    "build": "vue-cli-service build --no-unsafe-inline --skip-plugins pwa",
-    "buildsentry": "vue-cli-service build --no-unsafe-inline --skip-plugins pwa && chmod 755 tasks/sentry.sh && ./tasks/sentry.sh",
+    "build": "vue-cli-service build --no-unsafe-inline --skip-plugins pwa --mode production",
+    "buildsentry": "vue-cli-service build --no-unsafe-inline --skip-plugins pwa --modern --mode production && chmod 755 tasks/sentry.sh && ./tasks/sentry.sh",
```
Comment (Bitcoinera) [inline app/package.json:8]: I like modern 😎

File: app/src/components/common/TmSessionImport.vue [hunk 2]
```diff
@@ -61,14 +61,16 @@ import FieldSeed from "common/TmFieldSeed"
...
 import { mnemonicValidate } from "@polkadot/util-crypto"
 
 const has12or24words = (param) => {
   return (
     param && (param.split(` `).length === 12 || param.split(` `).length === 24)
   )
 }
+const isHex = value => {
+  return /0x[0-9a-f]+/i.test(value)
+}
```
Comment (Bitcoinera) [inline app/src/components/common/TmSessionImport.vue:73]: whhhhaaaa...? Are we leveling up and importing with our minds or what is going here

File: app/src/firebase.js [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+    if (firebaseInstance) {
+        return firebaseInstance
+    }
+    const [firebase] = await Promise.all([
+        import("firebase/app"),
+        import("firebase/auth")
+    ])
+    firebase.initializeApp(config.firebaseConfig)
+    firebaseInstance = firebase
+    return firebase
```
Comment (Bitcoinera) [inline app/src/firebase.js:15]:
```suggestion
try {
    const [firebase] = await Promise.all([
        import("firebase/app"),
        import("firebase/auth")
    ])
    firebase.initializeApp(config.firebaseConfig)
    firebaseInstance = firebase
    return firebase
   } catch (error) {
   console.error(`Firebase could not be initialized`)
   Sentry.captureException(error) // needs to be imported in an async way ✌️ 
   }
```

File: app/src/vuex/modules/account.js [hunk 4]
```diff
@@ -29,6 +27,7 @@ export default ({ apollo }) => {
 
   const actions = {
     async listenToAuthChanges({ commit }) {
+      const Auth = (await getFirebase()).auth()
```
Comment (Bitcoinera) [inline app/src/vuex/modules/account.js:30]: nice!

**Comment by Bitcoinera** [comment]
I'm testing it and it is working great! Bravo! 👏

**Pull request merged**